bokeh/core/property/text_like.py

```diff
@@ -42,6 +44,29 @@
 
 TextLike = Either(MathString, Instance("bokeh.models.text.MathText"), Instance("bokeh.models.text.PlainText"))
 
+class DictWithTextLikeValues(Dict):
+    """ Accept a Python dict with any keys and only TextLike values.
+    e.g.: Dict(key: Any, value: TextLike)
+    """
+    def validate(self, value, detail = False):
+        super().validate(value, detail=detail)
+
+        if isinstance(value, dict) and all(TextLike.is_valid(val) for _, val in value.items()):
```

Comment:
If you are hard-coding types like this, then what's the point of keeping value parameterization? In general this approach in unrobust and will require us to keep track of where we use `TextLike` and possibly other types equipped with transforms in future. Additionally this applies to `dict`, but does `list` (`Seq`), etc. support or should support  this as well?

Reply:
The point is that Dict does not call the `transform` methods of its values properties, and, `major_label_overrides` uses a `Dict` with property value of `TextLike` which solely purpose is to interpret string with the `transform` method into whatever `TextLike` model we have.

To make this broader, I could change it into `DictWithTransform` and remove the `TextLike`. Or make it more specific having only a property type `LabelOverrides`. What do you think?

Reply:
> Or make it more specific having only a property type LabelOverrides

FWIW this was my proposal:
```
major_label_overrides = LabelOverrides(help="...")
```
specifically so that we don't have to debate these policies (which, to date, have not ever been needed or requested) right now right before a release. If a second similar use case shows up we can figure out the appropriate generalization at that point (and use it to re-implement `LabelOverrides`)